cloudflare-analytics.sh

```diff
@@ -115,114 +130,393 @@ bandwidth=$bandwidth"
       echo "$series_name,$tags $data_points $cfTimeStamp" >> "$data_tmp"
     done
 
-    ${ECHO} curl -i -XPOST "$InfluxDBURL:$InfluxDBPort/write?precision=s&db=$InfluxDB" -u "$InfluxDBUser:$InfluxDBPassword" --data-binary @"$data_tmp"
+    post_influxdb_data_file "$data_tmp"
+}
 
-    # Clean up tmp file
-    rm -vf "${data_tmp}"
+#####################################################
+# Processing GraphQL results.
+function process_results_graphql {
+    local output="$1"
+
+    local keys=$(echo $output | jq keys[] | tr -d '"')
+    for k in $keys
+    do
+      local count=$(echo "$output" | jq .$k | jq length)
+      for i in $(seq -s ' ' 0 $((count-1)))
+      do
+        local values=$(echo $output | jq .$k[$i])
+        case $k in
+          httpRequests1mGroups)
+            process_request_graphql "$values"
+            ;;
+          firewallEventsAdaptive)
+            process_firewall_graphql "$values"
+            ;;
+          loadBalancingRequestsAdaptive)
+            ;;
+          *)
+            echo "process_requests: Unknown key: $k"
+            ;;
+        esac
+      done
+    done
 }
 
-function fetch_data {
-    local since=${1:-10080}
-    local until=${2:0}
+function process_request_graphql {
+    local output="$1"
+    local index=${2:-0}
+    local data_tmp="gql_$TMP_FILE_POSTFIX"
 
-    cloudflareRestURL="https://api.cloudflare.com/client/v4/zones/$cloudflarezoneid/analytics/dashboard?since=$since&until=$until&continuous=false"
-    output=$(curl -X GET "$cloudflareRestURL" \
-                            --header "Accept:application/json" \
-                            --header "Authorization:Bearer $cloudflareapikey" \
-                            2>&1 -k --silent)
-    process_results "$output"
+    ## Requests
+    cfRequestsAll=$(echo "$output" | jq --raw-output ".sum.requests")
+    cfRequestsCached=$(echo "$output" | jq --raw-output ".sum.cachedRequests")
+    cfRequestsUncached=$((cfRequestsAll-cfRequestsCached))
+    cfRequestsEncrypt=$(echo "$output" | jq --raw-output ".sum.encryptedRequests")
+
+    ## Bandwidth
+    cfBandwidthAll=$(echo "$output" | jq --raw-output ".sum.bytes")
+    cfBandwidthCached=$(echo "$output" | jq --raw-output ".sum.cachedBytes")
+    cfBandwidthUncached=$((cfBandwidthAll-cfBandwidthCached))
+    cfbandwidthEncrypt=$(echo "$output" | jq --raw-output ".sum.encryptedBytes")
+
+    ## Threats
+    cfThreatsAll=$(echo "$output" | jq --raw-output ".sum.threats")
+
+    ## Pageviews
+    cfPageviewsAll=$(echo "$output" | jq --raw-output ".sum.pageViews")
+
+    ## Unique visits
+    cfUniquesAll=$(echo "$output" | jq --raw-output ".uniq.uniques")
+
+    ## Timestamp
+    date=$(echo "$output" | jq --raw-output ".dimensions.datetime")
+    cfTimeStamp=$(date -d "${date}" '+%s')
+
+# The alignment here is important (DO NOT INDENT)
+    series_name="cloudflare_analytics"
```

Comment:
If this stuff below is essentially the same as line 121 - why repeat it?

Reply:
Well... whilst I see you point... this is in essence a 1 line string assignment. Creating a function to pass the params in just to assign a string seems excessive.

But also I imagine the 'api' variant will be removed when that api is deprecated (or sooner if we are happy with the new one).

Reply:
Resolved